### ISSUE TITLE
Update docker image to use more recent CouchDB

### ISSUE THREAD

**Issue by garrensmith** [pouchdb/pouchdb#6464]
We need to use a newer docker version of CouchDB. The current image `klaemo/couchdb:2.0-dev` is based off CouchDB 2.0. That is a bit old now and we have new features in CouchDB that PouchDB needs to test again.

**Comment by daleharvey** [comment]
Pinging @willholley since he has been doing master tests, so I think he is worried about us straight upgrading from our stable one as our tests have additional failures against master that they dont against 2.X and he has been working to get them green.

I think our plan could look something like ...

     1. Get current build green
     2. Switch the current 'couch-master' to be official 2.X stable (keep green)
     3. Add current master as failing
     4. Make master green
     5. Decide between 2.X or master and make that the default
     6. Migrate away any old tests (eventually remove isCouchMaster switches in the tests)

All sound good?

**Comment by willholley** [comment]
https://github.com/pouchdb/pouchdb/pull/6344 basically does 1-5 above - I need to rebase it in light of the recent fixes in master. The remaining piece is getting the mapreduce tests to pass against 2.x, but I'm leaning towards making that a separate PR as it's quite a lot of work and keeping those specifically on 1.x for now.

**Comment by garrensmith** [comment]
That is good with me. It just means I can't merge #6455 until we have the latest CouchDB master. But that isn't a problem.

**Comment by willholley** [comment]
@garrensmith #6455 is a prereq for using Couch master as the default so can you merge your changes and then I'll rebase my PR?

**Comment by garrensmith** [comment]
Ok great. I’ll do that on Monday. I just need to get the code coverage passing.
